samila/genimage.py

```diff
@@ -75,6 +75,8 @@ def generate(
             random.seed(self.seed)
             self.data1.append(self.function1(item[0], item[1]))
             self.data2.append(self.function2(item[0], item[1]))
+        self.data1 = list(map(lambda x: x.real, self.data1))
```

Comment:
This implementation is correct but it's not efficient.
It's better to filter the imaginary part in the previous loop (before appending)